### PR TITLE
EOS-24432: Provisioner kubernetes enviornment support for LC deployment

### PR DESCRIPTION
**Pull request by tanujashinde0405**
Signed-off-by: Tanuja Shinde <tanuja.shinde@seagate.com>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: test/deploy/k8s/cortx-pod-data.yml [hunk 1]
```diff
@@ -1,39 +1,55 @@
-apiVersion: v1
-kind: Pod
+apiVersion: apps/v1
```
Comment (ujjwalpl) [inline test/deploy/k8s/cortx-pod-data.yml:1]: Add these changes to a new file named cortx-deployment.yml

Comment (ujjwalpl) [reply test/deploy/k8s/cortx-pod-data.yml:1]: Also change deploy.sh to call the new file.

Comment (ujjwalpl) [reply test/deploy/k8s/cortx-pod-data.yml:1]: Also change deploy.sh to call the new file.

File: test/deploy/k8s/deploy.sh [hunk 2]
```diff
@@ -6,7 +6,7 @@ SCRIPT_DIR=$(dirname $0)
 
 kubectl create namespace cortx
 kubectl create cm solution-config --from-file="$SCRIPT_DIR"/cluster.yaml \
-    --from-file="$SCRIPT_DIR"/config.yaml --from-literal "node-id=123" --namespace cortx
+    --from-file="$SCRIPT_DIR"/config.yaml --from-literal "machine-id=3876243" --namespace cortx
```
Comment (ujjwalpl) [inline test/deploy/k8s/deploy.sh:9]: We should separate the cm for machine id, i.e. 

kubectl create cm solution-config --from-file="$SCRIPT_DIR"/cluster.yaml \
    --from-file="$SCRIPT_DIR"/config.yaml --namespace cortx

kubectl create cm node-1 --from-literal "machine-id=3876243" --namespace cortx

And use it to mount inside /etc/cortx/solution/node/ then create a link 
ln -sf /etc/cortx/solution/node/machine-id /etc/machine-id

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: test/deploy/k8s/cortx-pod-data.yml (hunk 1 above)
Comment (tanujashinde0405) [reply test/deploy/k8s/cortx-pod-data.yml]: done

File: test/deploy/k8s/deploy.sh (hunk 2 above)
Comment (tanujashinde0405) [reply test/deploy/k8s/deploy.sh]: reverted it to node-id itself as of now. will handle machine-id implementation in separate PR

**Review by ujjwalpl — APPROVED** [review]

**Review by ujjwalpl — APPROVED** [review]

**Review by 83bhp — APPROVED** [review]

**Pull request merged**